src/db_adapters/access_keys.rs

```diff
@@ -0,0 +1,141 @@
+                        {
+                            existing_record.deleted_by_receipt_id =
+                                Some(receipt.receipt_id.to_string());
+                        } else {
+                            access_keys.insert(
+                                (public_key.to_string(), receipt.receiver_id.to_string()),
+                                models::access_keys::AccessKey {
+                                    public_key: public_key.to_string(),
+                                    account_id: receipt.receiver_id.to_string(),
+                                    created_by_receipt_id: None,
+                                    deleted_by_receipt_id: Some(receipt.receipt_id.to_string()),
+                                    permission: models::enums::AccessKeyPermission::NotApplicable,
+                                },
+                            );
+                        }
+                    }
+                    _ => continue,
+                }
+            }
+        }
+    }
+
+    let mut access_keys_to_insert: Vec<models::access_keys::AccessKey> = vec![];
+
+    for (_, value) in access_keys {
+        if value.created_by_receipt_id.is_none() {
+            let target = schema::access_keys::table
+                .filter(schema::access_keys::dsl::public_key.eq(value.public_key))
+                .filter(schema::access_keys::dsl::account_id.eq(value.account_id));
+            loop {
+                match diesel::update(target.clone())
+                    .set(
+                        schema::access_keys::dsl::deleted_by_receipt_id
+                            .eq(value.deleted_by_receipt_id.clone()),
+                    )
+                    .execute_async(&pool)
+                    .await
+                {
+                    Ok(_) => break,
+                    Err(async_error) => {
+                        error!(
+                            target: crate::INDEXER_FOR_EXPLORER,
+                            "Error occurred while updating AccessKey. Retrying in {} milliseconds... \n {:#?}",
+                            crate::INTERVAL.as_millis(),
+                            async_error,
+                        );
+                        tokio::time::delay_for(crate::INTERVAL).await;
+                    }
+                }
+            }
+        } else {
+            access_keys_to_insert.push(value);
+        }
+    }
+
+    loop {
+        match diesel::insert_into(schema::access_keys::table)
+            .values(access_keys_to_insert.clone())
+            .on_conflict((schema::access_keys::dsl::public_key, schema::access_keys::dsl::account_id))
+            .do_update()
+            .set((
+                schema::access_keys::dsl::created_by_receipt_id
+                    .eq(excluded(schema::access_keys::dsl::created_by_receipt_id)),
+                schema::access_keys::dsl::deleted_by_receipt_id
+                    .eq(excluded(schema::access_keys::dsl::deleted_by_receipt_id)),
```

Comment:
We should also override `permission_kind`. Is there a way to just say that we want to override all the fields?